### DISCUSSION TITLE
How to set up KeystoneJS in a yarn workspace setup?

### DISCUSSION THREAD

**Discussion by tyteen4a03** [keystonejs/keystone#7480] [Questions] [answered]
I'm trying to set up Keystone in an existing yarn workspace setup but I keep getting errors like this:

```
➤ YN0001: │ Error: While cloning  website/packages/web-panel/node_modules/@keystone-6/core/node_modules/@types/react -> website/packages/web-panel/node_modules/@keystone-6/fields-document/node_modules/@types/react ENOENT: no such file or directory, scandir website/packages/web-panel/node_modules/@keystone-6/core/node_modules/@types/react'
```

At other times when using `create-keystone-app` it claims that it can't find `react` when installing into `packages/web-panel`.

Are there tutorials on how to create a new KeystoneJS app inside an existing workspace?

**Comment by daawaan4x** [comment]
Hi @tyteen4a03, what OS do you use for development? I had issues with trying to setup keystone in Windows but it worked in a Linux environment

**Reply by tyteen4a03** [reply]
I use macOS 12.

**Reply by daawaan4x** [reply]
@tyteen4a03 try to set it up in linux, you can rent a small virtual machine on cloud platform providers for development

**Comment by raveling** [comment, marked as answer]
@tyteen4a03 Can you please provide a more complete reproduction for the error?

> Are there tutorials on how to create a new KeystoneJS app inside an existing workspace?

We don't have dedicated tutorials, but there's nothing Keystone specific you need to consider when setting up your own yarn workspaces.
